cl/sentinel/sentinel.go

```diff
@@ -325,6 +329,89 @@ func (s *Sentinel) HasTooManyPeers() bool {
 	return active >= peers.DefaultMaxPeers
 }
 
+func (s *Sentinel) isPeerUsefulForAnySubnet(node *enode.Node) bool {
+	ret := false
+
+	nodeAttnets := bitfield.NewBitvector64()
+	nodeSyncnets := bitfield.NewBitvector4()
+	if err := node.Load(enr.WithEntry(s.cfg.NetworkConfig.AttSubnetKey, &nodeAttnets)); err != nil {
+		log.Trace("Could not load att subnet", "err", err)
+		return false
+	}
+	if err := node.Load(enr.WithEntry(s.cfg.NetworkConfig.SyncCommsSubnetKey, &nodeSyncnets)); err != nil {
+		log.Trace("Could not load sync subnet", "err", err)
+		return false
+	}
+
+	s.subManager.subscriptions.Range(func(key, value any) bool {
+		sub := value.(*GossipSubscription)
+		if sub.sub == nil {
+			return true
+		}
+
+		if !sub.subscribed.Load() {
+			return true
+		}
+
+		if len(sub.topic.ListPeers()) > peerSubnetTarget {
+			return true
+		}
+		if gossip.IsTopicBeaconAttestation(sub.sub.Topic()) {
+			ret = s.isPeerUsefulForAttNet(sub, nodeAttnets)
```

Comment:
I feel the logic can be more explicit so the code turns to be readable, but feel free to see if you'd like to take my advise.
```
if peerUseful = s.isPeerUsefulForAttNet(sub, nodeAttnets); peerUseful {
  return false
}
return true
```